src/main/cpp/src/cast_string.cu

```diff
@@ -176,7 +584,59 @@ struct row_valid_fn {
   bitmask_type const* string_col_null_mask;
 };
 
+/**
+ * @brief validates a column has no ansi errors, throws if error
+ *
+ * @param col column to verify
+ * @param source_col original string column used to create col
+ * @param stream stream on which to operate
+ */
+void validate_ansi_column(column_view const& col,
+                          strings_column_view const& source_col,
+                          rmm::cuda_stream_view stream)
+{
+  auto const num_nulls      = col.null_count();
+  auto const incoming_nulls = source_col.null_count();
+  auto const num_errors     = num_nulls - incoming_nulls;
+  if (num_errors > 0) {
+    auto const first_error = thrust::find_if(rmm::exec_policy(stream),
```

Comment:
What if `first_error` iter is out of bound?

Reply:
if `num_errors` > 0 then `first_error` will be valid. I can check this anyway, but it would be a sanity check.